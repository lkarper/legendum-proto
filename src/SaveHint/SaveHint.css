@import url('https://fonts.googleapis.com/css2?family=Cinzel+Decorative:wght@400;700&family=Crimson+Text&display=swap');

.SaveHint__submit {
    margin: 5px;
    max-width: 200px;
}

.SaveHint__form {
    margin: 5px;
}

.SaveHint__form textarea {
    font-family: 'Crimson Text', serif;    
    font-size: 16px;
}

@media screen and (min-width: 1000px) {
    .SaveHint__form {
        display: flex;
        flex-direction: column;
        width: 800px;
    }

    .SaveHint__submit {
        align-self: center;
    }

    #custom-note {
        height: 200px;
        font-size: 18px;
    }
}