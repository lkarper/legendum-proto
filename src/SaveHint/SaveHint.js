import React, { useState } from 'react';
import ValidateCustomNote from '../ValidateCustomNote/ValidateCustomNote';

const SaveHint = (props) => {

    const [customNote, setCustomNote] = useState('');
    const [customNoteError, setCustomNoteError] = useState('');

    const { onSubmitHint } = props;

    return (
        <div>
            <form onSubmit={(e) => onSubmitHint(e, customNote)}>
                <textarea
                    id="custom-note"
                    name="custom-note"
                    aria-label="Add a custom note here, if you'd like."
                    placeholder="Add a custom note here, if you'd like."
                    rows="4"
                    cols="50"
                    value={customNote}
                    onChange={(e) => setCustomNote(e.target.value)}
                ></textarea>
                <button 
                    type="submit"
                    disabled={!!customNoteError}
                >Save Hint</button>
                <ValidateCustomNote 
                    customNote={customNote}
                    customNoteError={customNoteError}
                    setCustomNoteError={setCustomNoteError}
                />
            </form>
        </div>
    )
}

export default SaveHint;