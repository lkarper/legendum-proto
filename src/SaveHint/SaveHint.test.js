import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import SaveHint from './SaveHint';

describe('SaveHint component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<SaveHint />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<SaveHint />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
