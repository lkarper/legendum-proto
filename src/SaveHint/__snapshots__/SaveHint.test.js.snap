// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveHint component renders the UI as expected 1`] = `
<SaveHint
  cNoteProp=""
  onSubmitHint={[Function]}
>
  <form
    className="SaveHint__form"
    onSubmit={[Function]}
  >
    <textarea
      aria-label="Add a custom note here, if you'd like."
      className="SaveHint__textarea"
      id="custom-note"
      name="custom-note"
      onChange={[Function]}
      placeholder="Add a custom note here, if you'd like."
      rows="4"
      value=""
    />
    <button
      className="SaveHint__submit button"
      disabled={false}
      type="submit"
    >
      Save Hint
    </button>
    <ValidateCustomNote
      customNote=""
      customNoteError=""
      setCustomNoteError={[Function]}
    >
      <div
        role="alert"
      />
    </ValidateCustomNote>
  </form>
</SaveHint>
`;
